### ISSUE TITLE
NEXUS-37518 Failed to run task 'Docker - Delete unused manifests and images' version 3.41.0-01

### ISSUE THREAD

**Issue by danischroeter** [sonatype/nexus-public#142]
clone of https://issues.sonatype.org/browse/NEXUS-37518 to not miss this important issue...

Hi,

We are seeing an issue reproducible in several nexus instances we have running on version 3.41.0-01 when running the task Remove unused docker manifests and images (Docker - Delete unused manifests and images ). The task fails with the bellow logs.
```
2023-02-16 10:58:38,340+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.tasks.DockerGCTask - Task information:
2023-02-16 10:58:38,341+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.tasks.DockerGCTask -  ID: 26011f0c-640a-4268-aa8a-9089d3d65122
2023-02-16 10:58:38,342+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.tasks.DockerGCTask -  Type: repository.docker.gc
2023-02-16 10:58:38,342+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.tasks.DockerGCTask -  Name: Remove unused docker manifests and images
2023-02-16 10:58:38,342+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.tasks.DockerGCTask -  Description: Docker - Delete unused manifests and images
2023-02-16 10:58:38,344+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl - Garbage collection starting on repository: RepositoryImpl$$EnhancerByGuice$$677596588{type=proxy, format=docker, name='dockerhub-proxy'}
2023-02-16 10:58:38,346+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl - Running GC for v1 assets on dockerhub-proxy
2023-02-16 10:58:38,356+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl - Finished GC for v1 assets on dockerhub-proxy. checked 0, deleted 0
2023-02-16 10:58:38,356+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl - Running GC for v2 assets on dockerhub-proxy
2023-02-16 10:58:38,357+0000 INFO  [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl - Fetched data for RepositoryImpl$$EnhancerByGuice$$677596588{type=proxy, format=docker, name='dockerhub-proxy'}. checked 0
2023-02-16 10:58:38,372+0000 ERROR [quartz-12-thread-5]  *SYSTEM org.sonatype.nexus.repository.docker.tasks.DockerGCTask - Failed to run task 'Docker - Delete unused manifests and images' on repository 'dockerhub-proxy'
java.lang.NullPointerException: Cannot get property 'digest' on null object
    at org.codehaus.groovy.runtime.NullObject.getProperty(NullObject.java:60)
    at org.codehaus.groovy.runtime.InvokerHelper.getProperty(InvokerHelper.java:190)
    at org.codehaus.groovy.runtime.DefaultGroovyMethods.getAt(DefaultGroovyMethods.java:342)
    at org.codehaus.groovy.runtime.dgm$242.invoke(Unknown Source)
    at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoMetaMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:251)
    at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
    at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
    at org.codehaus.groovy.runtime.callsite.NullCallSite.call(NullCallSite.java:34)
    at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:116)
    at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
    at org.sonatype.nexus.repository.docker.internal.V2Manifest.referencedDigests(V2Manifest.groovy:104)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.fetchDataForManifest(DockerGCFacetImpl.java:382)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.fetchDataForRepository(DockerGCFacetImpl.java:355)
    at org.sonatype.nexus.transaction.TransactionalWrapper.proceedWithTransaction(TransactionalWrapper.java:58)
    at org.sonatype.nexus.transaction.TransactionInterceptor.proceedWithTransaction(TransactionInterceptor.java:66)
    at org.sonatype.nexus.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:55)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.getExternalUsedLayersDigests(DockerGCFacetImpl.java:267)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.checkForReferencesInOtherRepositories(DockerGCFacetImpl.java:246)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.handleV2Assets(DockerGCFacetImpl.java:221)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.processRepository(DockerGCFacetImpl.java:144)
    at org.sonatype.nexus.transaction.TransactionalWrapper.proceedWithTransaction(TransactionalWrapper.java:58)
    at org.sonatype.nexus.transaction.TransactionInterceptor.proceedWithTransaction(TransactionInterceptor.java:66)
    at org.sonatype.nexus.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:55)
    at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.deleteUnusedManifestsAndImages(DockerGCFacetImpl.java:126)
    at org.sonatype.nexus.common.stateguard.MethodInvocationAction.run(MethodInvocationAction.java:39)
    at org.sonatype.nexus.common.stateguard.StateGuard$GuardImpl.run(StateGuard.java:272)
    at org.sonatype.nexus.common.stateguard.GuardedInterceptor.invoke(GuardedInterceptor.java:54)
    at org.sonatype.nexus.repository.docker.tasks.DockerGCTask.execute(DockerGCTask.java:48)
    at org.sonatype.nexus.repository.RepositoryTaskSupport.execute(RepositoryTaskSupport.java:79)
    at org.sonatype.nexus.scheduling.TaskSupport.call(TaskSupport.java:100)
    at org.sonatype.nexus.quartz.internal.task.QuartzTaskJob.doExecute(QuartzTaskJob.java:143)
    at org.sonatype.nexus.quartz.internal.task.QuartzTaskJob.execute(QuartzTaskJob.java:106)
    at org.quartz.core.JobRunShell.run(JobRunShell.java:202)
    at org.sonatype.nexus.quartz.internal.QuartzThreadPool.lambda$0(QuartzThreadPool.java:145)
    at org.sonatype.nexus.thread.internal.MDCAwareRunnable.run(MDCAwareRunnable.java:40)
    at org.apache.shiro.subject.support.SubjectRunnable.doRun(SubjectRunnable.java:120)
    at org.apache.shiro.subject.support.SubjectRunnable.run(SubjectRunnable.java:108)
    at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
    at java.util.concurrent.FutureTask.run(FutureTask.java:266)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
    at java.lang.Thread.run(Thread.java:750) 

 The problematic image/manifest has oci mediatype.

The only images available on the repository belonged to scylladb/scylla and the issue was reproduce having the tag latest and 5.0.rc8 

These images are public and can be pulled from docker hub:
```
 

`docker pull scylladb/scylla:5.0.rc8`

 

 

`docker pull scylladb/scylla:latest`

the problematic manifest seems to belong to the tag 5.0.rc8 (which has a oci mediatype)

Logs and print screens attached:

    Screenshot 2023-02-16 at 10.53.23 (images available on the repository)

I hope this is enough evidence to be able to be reproducible.

**Comment by danischroeter** [comment]
This still happens also with the latest 3.53.3

I also tried to enable more debug logging to at least find out what causes the issue - but to no avail.
Does anybody have an idea how to find the artifacts causing the issue?

Extending the log message with the docker image causing this would be a quick/easy improvement that would help to manually cleanup until properly fixed - and also for future issues...
```
    Suppressed: java.lang.NullPointerException: Cannot get property 'digest' on null object
        at org.codehaus.groovy.runtime.NullObject.getProperty(NullObject.java:60)
        at org.codehaus.groovy.runtime.InvokerHelper.getProperty(InvokerHelper.java:190)
        at org.codehaus.groovy.runtime.DefaultGroovyMethods.getAt(DefaultGroovyMethods.java:342)
        at org.codehaus.groovy.runtime.dgm$242.invoke(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoMetaMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:251)
        at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.call(PogoMetaMethodSite.java:71)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.NullCallSite.call(NullCallSite.java:34)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.PojoMetaMethodSite.call(PojoMetaMethodSite.java:58)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:128)
        at org.sonatype.nexus.repository.docker.internal.V2Manifest.referencedDigests(V2Manifest.groovy:104)
        at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.fetchDataForManifest(DockerGCFacetImpl.java:382)
        at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.fetchDataForRepository(DockerGCFacetImpl.java:355)
        at org.sonatype.nexus.transaction.TransactionalWrapper.proceedWithTransaction(TransactionalWrapper.java:58)
        at org.sonatype.nexus.transaction.TransactionInterceptor.proceedWithTransaction(TransactionInterceptor.java:66)
        at org.sonatype.nexus.transaction.TransactionInterceptor.invoke(TransactionInterceptor.java:55)
        at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.getExternalUsedLayersDigests(DockerGCFacetImpl.java:267)
        at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.checkForReferencesInOtherRepositories(DockerGCFacetImpl.java:246)
        at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.handleV2Assets(DockerGCFacetImpl.java:221)
        at org.sonatype.nexus.repository.docker.internal.orient.DockerGCFacetImpl.processRepository(DockerGCFacetImpl.java:144)

**Comment by gracecllee** [comment]
This issue has been fixed in Release 3.54.0.

**Comment by gracecllee** [comment]
This issue has been fixed in Release 3.54.0. Please see our release note: https://help.sonatype.com/repomanager3/product-information/release-notes/2023-release-notes/sonatype-nexus-repository-3.54.0---3.54.1-release-notes
